src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -1766,4 +1724,140 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
+        const relatedRecordFieldName =
+            relatedRecordFieldNameFor(this.lookupFieldApiNameFor(recordId));
+        this.updateFormState({[relatedRecordFieldName]: record});
+    }
+
+    lookupFieldApiNameFor(recordId) {
+        return this.lookupFieldApiNameBySelectedRecordId[recordId];
+    }
+
+    updateFormState(fields) {
+        Object.assign(this.formState, fields);
+
+        if (this.hasImportedRecordFieldsBeingSetToNull(fields)) {
+            this.deleteRelationshipFieldsFromStateFor(fields);
+        }
+
+        // Re-assign to prompt reactivity
+        this.formState = JSON.parse(JSON.stringify(this.formState));
+    }
+
+    hasImportedRecordFieldsBeingSetToNull(fields) {
+        return Object.keys(fields)
+            .filter(field =>
+                GeFormService.importedRecordFieldNames.includes(field) &&
+                fields[field].value === null
+            ).length > 0;
+    }
+
+    deleteRelationshipFieldsFromStateFor(fields) {
+        const needsRelationshipFieldDeleted = (field) =>
+            this.hasRelatedRecordFieldInFormState(field) &&
+            fields[field].value === null;
+
+        Object.keys(fields)
+            .filter(needsRelationshipFieldDeleted)
+            .forEach(field => {
+                this.deleteFieldFromFormState(relatedRecordFieldNameFor(field));
+            });
+    }
+
+    deleteFieldFromFormState(field) {
+        delete this.formState[field];
+    }
+
+    hasRelatedRecordFieldInFormState(field) {
+        return this.formState.hasOwnProperty(relatedRecordFieldNameFor(field));
+    }
+
+    handleFormFieldChange(event) {
+        //handleChangeDonationDonor logic moved here from geFormField
+        if (event.detail.hasOwnProperty(DATA_IMPORT_DONATION_DONOR_FIELD.fieldApiName)) {
+            //todo: eventually this should just be a reaction to formState changing
+            this.handleDonationDonorChange(event.detail[DATA_IMPORT_DONATION_DONOR_FIELD.fieldApiName])
+        }
+        this.updateFormState(event.detail);
+    }
+
+    initializeState() {
+        if (this.sections) {
+            this.sections
+                .forEach(section => {
+                    if (section.elements) {
+                        section.elements
+                            .forEach(element => {
+                                this.setInitialValueInFormStateForElement(element);
+                            });
+                    }
+                });
+        }
+    }
+
+    setInitialValueInFormStateForElement(element) {
+        this.updateFormState(
+            {
+                [this.getSourceFieldApiNameFor(element)]: {value: this.getValueFrom(element)}
+            }
+        );
+    }
+
+    setInitialValueInFormStateForFieldMappings(fieldMappingDevNames) {
+        this.elementsFor(fieldMappingDevNames)
+            .forEach(el => this.setInitialValueInFormStateForElement(el));
+
+    }
+
+    elementsFor(fieldMappingDevNames) {
```

Comment:
This is returning `undefined` and is throwing a component error when I clear a lookup to the donor account/contact.

Reply:
Added the return statement into the function here as we discussed and should be working again now.